features/step_definitions/helper_services.rb

```diff
@@ -120,19 +120,32 @@
 #save the service ip and port of the proxy pod for later use in the scenario.
 Given /^I have a(n authenticated)? proxy configured in the project$/ do |use_auth|
   if use_auth
-    step %Q/I run the :new_app client command with:/, table(%{
-      | docker_image | aosqe/squid-proxy  |
-      | env          | USE_AUTH=1         |
+    step %Q/I run the :create_deploymentconfig client command with:/, table(%{
+      | image | aosqe/squid-proxy  |
+      | name  | squid-proxy        |
+      })
+    step %Q/I wait until the status of deployment "squid-proxy" becomes :running/
+    step %Q/I run the :set_env client command with:/, table(%{
+      | resource | deploymentconfig/squid-proxy |
+      | e        | USE_AUTH=1                   |
+      })
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | deployment=squid-proxy-2 |
       })
   else
-    step %Q/I run the :new_app client command with:/, table(%{
-      | docker_image | aosqe/squid-proxy  |
+    step %Q/I run the :create_deployment client command with:/, table(%{
+      | image | aosqe/squid-proxy  |
+      | name  | squid-proxy        |
+      })
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | deployment=squid-proxy-1 |
       })
   end
   step %Q/the step should succeed/
-  step %Q/a pod becomes ready with labels:/, table(%{
-    | deployment=squid-proxy-1 |
-    })
+  @result = user.cli_exec(:expose, resource: "deploymentconfig", resource_name: "squid-proxy", port: "3128")
+  unless @result[:success]
+    raise "could not create squid-proxy service, see log"
```

Comment:
@zhouying7780 to better help debug, perhaps you can print out @result[:response] as well?

Reply:
It is printed by itself unless `quiet` flag has been set. And I don't see it set.